am/scripted-decision/src/get-location-from-ip-address.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @file Determine the authenticating user location
+ * based on their IP address
+ * and save the location information in the sharedState object
+ * in a Scripted Decision Node script
+ * in ForgeRock Access Management (AM).
+ * @see {@link https://backstage.forgerock.com/docs/am/7.1/authentication-guide/scripting-api-node.html}.
+ * @version 0.1.0
+ */
+
+/**
+ * Use the function scope to avoid inconsistent behavior
+ * that may occur in the top-level scope in AM Rhino.
+ */
+(function () {
+    /**
+     * An example definition for the object containing the location information.
+     * @typedef {object} LoginLocation
+     * @property {number} longitude
+     * @property {number} latitude
+     * @property {string} country The country code.
+     * @example
+     * {
+     *     "longitude": -0.1585557,
+     *     "latitude": 51.523767,
+     *     "country": "GB"
+     * }
+     */
+
+    /**
+     * Derive the location information from an IP address by using the IP Geolocation API:
+     * {@link https://ip-api.com/}.
+     * For using this API over HTTPS, you will need to sign up for key.
```

Comment:
Suggestion: ...sign up for a key.

Reply:
Line 34: Wondering: ...other providers providing (can also use "offering")

Reply:
Yeah, that's what providers do 🙂 
Good catch, thanks!